component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -170,6 +180,47 @@ protected void processAuthenticationResponse(HttpServletRequest httpServletReque
         }
     }
 
+    private String getMatchedSubjectAttribute(String certAttributes, String subjectAttributePattern,
+            AuthenticationContext authenticationContext) throws AuthenticationFailedException {
+
+        LdapName ldapDN;
+        try {
+            ldapDN = new LdapName(certAttributes);
+        } catch (InvalidNameException e) {
+            throw new AuthenticationFailedException("error occurred while get the certificate claims", e);
+        }
+        if (subjectPatternCompiled == null) {
+
+            subjectPatternCompiled = Pattern.compile(subjectAttributePattern);
+        }
+        String userNameAttribute = getAuthenticatorConfig().getParameterMap().get(X509CertificateConstants.USERNAME);
+        List<String> matches = new ArrayList<>();
```

Comment:
Use a different name. This confuses with regex matcher. ex:- matchedStringList

Reply:
done